### PR TITLE
Adapt docs to App Router update and add Foundry use case to verify section

### LINKED ISSUE #53: App router migration changes
### Description : 

We recently migrated to Next `app` router, lets update the references inline with it : 

Some things top of my mind : 

1. Updating min node version in [Installation guide](https://docs.scaffoldeth.io/quick-start/installation) we updated it in [SE-2 README](https://github.com/scaffold-eth/scaffold-eth-2?tab=readme-ov-file#requirements)

2. Updating frontend path to => `packages/nextjs/app` in [Environment#Whats_Next](https://docs.scaffoldeth.io/quick-start/environment#whats-next)

### LINKED ISSUE #55: Adapt contract verify section to add Foundry details
We've just updated contract verify section, but we should adapt it to cover Foundry use case also. 

Maybe we could split the section in 2, leaving current explanations for Hardhat subsection/bullet point, and adding the `yarn verify" for Foundry, explaining which command it's actually using.

### PR DESCRIPTION
**Pull request by Pabl0cks**
Adding these two changes in the same PR, they are not related but are small enough to not create 2 PRs I think:

- Closes #53 
- Closes #55. Here I could have added a reference to `yarn deploy:verify` but maybe just adds complexity to the section. What do you think?

Regarding [Documenting folder structure](https://github.com/scaffold-eth/scaffold-eth-2/pull/535#issuecomment-1879794609) I think we could document the components structure in `/components` category description. Another possibility would be adding a new bullet point in "What's Next"  from `/quick-start/environment`, with something like "Add new components (if you need any) in...".

Let me know what do you think, thanks! 🙌

### CODE REVIEW

File: docs/quick-start/environment.mdx [hunk 1]
```diff
@@ -56,7 +56,7 @@ Visit your app on `http://localhost:3000`. You can interact with your smart cont
 - Edit your deployment scripts:
   - Hardhat => `packages/hardhat/deploy`
   - Foundry => `packages/foundry/script`
-- Edit your frontend in `packages/nextjs/pages`
+- Edit your frontend in `packages/nextjs/app/page.tsx` (homepage), and the rest of [pages / layouts](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts) for the different [routes](https://nextjs.org/docs/app/building-your-application/routing/defining-routes) of your application.
```
Comment (technophile-04) [inline docs/quick-start/environment.mdx:59]: "and the rest of page / layout ..." seems a bit confusing to me 😅 

Maybe : 

"Edit your frontend homepage at `packages/nextjs/app/page.tsx`,  checkout next docs on how [routing](https://nextjs.org/docs/app/building-your-application/routing/defining-routes) and configuring [pages / layout](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts) works"

Lol not sure if I made it more complicated. but please feel free to update / improve it🙌

**Comment by technophile-04** [comment]
Tysm @Pabl0cks !! Just added a small comment but other than that everything looks good !!

File: docs/quick-start/environment.mdx (hunk 1 above)
Comment (Pabl0cks) [reply docs/quick-start/environment.mdx]: Love your suggestions, just pushed the change! 🙌

File: docs/deploying/deploy-smart-contracts.mdx [hunk 2]
```diff
@@ -60,9 +60,12 @@ yarn verify --network network_name
...
 eg: `yarn verify --network sepolia`
 
-This uses [etherscan-verify from hardhat-deploy](https://www.npmjs.com/package/hardhat-deploy#4-hardhat-etherscan-verify) to verify all the deployed contracts.
+This command **works in both Hardhat and Foundry**, verifying all the deployed contracts. The method of verification varies within the solidity framework:
 
-You can alternatively use [hardhat-verify](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify) to verify your contracts, passing network name, contract address and constructor arguments (if any): `yarn hardhat-verify --network network_name contract_address "Constructor arg 1"`
+- Hardhat => uses [etherscan-verify from hardhat-deploy](https://www.npmjs.com/package/hardhat-deploy#4-hardhat-etherscan-verify).
+- Foundry => uses `VerifyAll.s.sol` script located in `packages/foundry/script`.
+
+**In Hardhat**, you can alternatively use [hardhat-verify](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify) to verify your contracts, passing network name, contract address and constructor arguments (if any): `yarn hardhat-verify --network network_name contract_address "Constructor arg 1"`
```
[the remark this answers is not in the document]
Comment (Pabl0cks) [reply docs/deploying/deploy-smart-contracts.mdx]: Awesome suggestions! Just pushed them all 💪

**Comment by Pabl0cks** [comment]
Thanks for the review a suggestions @technophile-04 and @edakturk14 !! I've just pushed the changes 🙌

**Review by technophile-04 — APPROVED** [review]
Tysm all !! Merging this 🙌

**Pull request merged**